Core.cpp

```diff
@@ -45,10 +48,18 @@ int main(int, char**)
 
 		if (bboxes.size() && !flag) {
 			//High Beam detected
+
+			detected_image = frame; //save detected frame;
```

Comment:
We'll only save it when we enter the `if` block, that is, only on detection. It'll be the original unaltered frame. We only need to keep a copy of every unaltered frame, and it'll only be saved on detection. Right now, you've frame, original_frame, detected_frame. We could instead simply do `detected_frame = frame` above and and then use that directly in `imwrite`.